docs: update localization docs and modernize examples

SkyZeroZx opened the pull request:
Improves the localization documentation by updating code highlight syntax issues and removing outdated TODOs.

Additionally, the CDK examples have been updated to use the viewChild query signal API, aligning the docs with current recommended patterns.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A

## What is the new behavior?

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

JeanMeche commented on adev/src/content/examples/drag-drop/src/root-element/app/app.component.ts:
```diff
@@ -23,13 +23,12 @@ import {
   private _viewContainerRef = inject(ViewContainerRef);
-
-  @ViewChild(TemplateRef) _dialogTemplate!: TemplateRef<any>;
+  _dialogTemplate = viewChild.required(TemplateRef);
```
nit: Could you also make it private, it's unused in the template (and also that was hinted by the leading underscore. 
```suggestion
  private _dialogTemplate = viewChild.required(TemplateRef);
```

SkyZeroZx committed:
docs: update localization docs and modernize examples

Improves the localization documentation by updating code highlight syntax issues
and removing outdated TODOs.

Additionally, the CDK examples have been updated to use the viewChild query signal API,
aligning the docs with current recommended patterns.
